spec.md

```diff
@@ -202,11 +202,11 @@ help intermediate gateways determine how to route the events.
 
 ### source
 * Type: `URI-reference`
-* Description: This describes the event producer. Often this will include
-  information such as the type of the event source, the organization
-  publishing the event, the process that produced the event, and some unique
-  identifiers. The exact syntax and semantics behind the data encoded in the URI
-  is event producer defined.
+* Description: Identifies the event producer. If the source includes a
+  URI authority [RFC3986  §3.2](https://tools.ietf.org/html/rfc3986#section-3.2)
+  then it SHOULD be a valid, internet-unique URI for the producer. If there is no
```

Comment:
I'm not sure what `valid` is supposed to say. It MUST be valid according to RFC3986 already.
If you mean valid as in "I should be able to make a request towards this URI, and end up at the producer" - I don't think this is a good idea. Take any GraphQL API.
I'm taking GitHub as the example, assume for a second that there is no web interface. `https://api.github.com/graphql` is then the only valid URL, but it would be a valid choice to have `https://api.github.com/cloudevents/spec/123` for the `source`.

Reply:
"valid" is shoddy, will reword. Definitely did not mean to imply a URL.